borg/helpers.py

```diff
@@ -885,9 +891,9 @@ def canonical_path(self):
             return self.path
         else:
             if self.path and self.path.startswith('~'):
-                path = '/' + self.path
+                path = '/' + self.path  # /~/x = path x relative to home dir
             elif self.path and not self.path.startswith('/'):
-                path = '/~/' + self.path
+                path = '/./' + self.path  # /./x = path x relative to cwd
```

Comment:
this will trigger a "repo was previously located at ..." warning once if a user uses a relative path, because the url stored in cache before (old style) is compared with different url generated by updated code (new style).